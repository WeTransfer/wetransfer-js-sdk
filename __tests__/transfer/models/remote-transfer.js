const { RemoteTransfer } = require('../../../src/transfer/models');

describe('RemoteTransfer model', () => {
  let transfer = null;
  beforeEach(() => {
    transfer = new RemoteTransfer({
      id: 'random-hash',
      version_identifier: null,
      state: 'uploading',
      shortened_url: 'https://we.tl/s-random-hash',
      name: 'Little kittens',
      description: 'Something about cats, most probably.',
      size: 0,
      total_items: 0,
      items: [],
    });
  });

  describe('contructor', () => {
    it('should create a normalized transfer', () => {
      expect(transfer).toMatchSnapshot();
    });
  });

  describe('when items are added to the transfer', () => {
    beforeEach(() => {
      const items = [
        {
          id: 'random-hash',
          name: 'kittie.gif',
          content_identifier: 'file',
        },
        {
          id: 'random-hash',
          name: 'wetransfer.com',
          content_identifier: 'web_content',
        },
      ];
      transfer.addItems(...items);
    });

    it('should add items to the transfer', () => {
      expect(transfer.items).toMatchSnapshot();
    });

    describe('files property', () => {
      it('should return only files', () => {
        expect(transfer.files).toMatchSnapshot();
      });
    });

    describe('links property', () => {
      it('should return only links', () => {
        expect(transfer.links).toMatchSnapshot();
      });
    });
  });
});
