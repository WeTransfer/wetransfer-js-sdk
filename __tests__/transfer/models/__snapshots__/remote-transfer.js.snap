// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RemoteTransfer model contructor should create a normalized transfer 1`] = `
RemoteTransfer {
  "description": "Something about cats, most probably.",
  "id": "random-hash",
  "items": Array [],
  "name": "Little kittens",
  "shortened_url": "https://we.tl/s-random-hash",
  "size": 0,
}
`;

exports[`RemoteTransfer model when items are added to the transfer files property should return only files 1`] = `
Array [
  Object {
    "content_identifier": "file",
    "id": "random-hash",
    "name": "kittie.gif",
    "transfer": RemoteTransfer {
      "description": "Something about cats, most probably.",
      "id": "random-hash",
      "items": Array [
        [Circular],
        Object {
          "content_identifier": "web_content",
          "id": "random-hash",
          "name": "wetransfer.com",
          "transfer": [Circular],
        },
      ],
      "name": "Little kittens",
      "shortened_url": "https://we.tl/s-random-hash",
      "size": 0,
    },
  },
]
`;

exports[`RemoteTransfer model when items are added to the transfer links property should return only links 1`] = `
Array [
  Object {
    "content_identifier": "web_content",
    "id": "random-hash",
    "name": "wetransfer.com",
    "transfer": RemoteTransfer {
      "description": "Something about cats, most probably.",
      "id": "random-hash",
      "items": Array [
        Object {
          "content_identifier": "file",
          "id": "random-hash",
          "name": "kittie.gif",
          "transfer": [Circular],
        },
        [Circular],
      ],
      "name": "Little kittens",
      "shortened_url": "https://we.tl/s-random-hash",
      "size": 0,
    },
  },
]
`;

exports[`RemoteTransfer model when items are added to the transfer should add items to the transfer 1`] = `
Array [
  Object {
    "content_identifier": "file",
    "id": "random-hash",
    "name": "kittie.gif",
    "transfer": RemoteTransfer {
      "description": "Something about cats, most probably.",
      "id": "random-hash",
      "items": [Circular],
      "name": "Little kittens",
      "shortened_url": "https://we.tl/s-random-hash",
      "size": 0,
    },
  },
  Object {
    "content_identifier": "web_content",
    "id": "random-hash",
    "name": "wetransfer.com",
    "transfer": RemoteTransfer {
      "description": "Something about cats, most probably.",
      "id": "random-hash",
      "items": [Circular],
      "name": "Little kittens",
      "shortened_url": "https://we.tl/s-random-hash",
      "size": 0,
    },
  },
]
`;
